base62.go

```diff
@@ -1,17 +1,17 @@
 package ksuid
 
-import "bytes"
+import "strings"
 
 // lexographic ordering (based on Unicode table) is 0-9A-Za-z
-var base62Characters = []byte("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz")
+var base62Characters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
 
-func base2base(in []byte, inBase int, outBase int) []byte {
-	res := []byte{}
-	bs := in[:]
+func appendBase2Base(dst []byte, src []byte, inBase int, outBase int) []byte {
+	bs := src[:]
+	bq := [stringEncodedLength]byte{}
```

Comment:
It doesn't violate the change, `bq` here is used as the initial backend array, but if `append` needs a larger space it'll make a dynamic memory allocation to support it.

Setting the initial array capacity to `stringEncodedLength` is just an optimization for the use case that we have.